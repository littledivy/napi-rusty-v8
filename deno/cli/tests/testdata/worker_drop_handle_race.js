// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.

// https://github.com/denoland/deno/issues/11342
// Test for a panic that happens when the main thread's event loop finishes
// running while the worker's event loop is still spinning.

// The exception thrown in the worker will not terminate the worker, but it will
// propagate to the main thread and cause it to exit.
new Worker(
  new URL("./workers/drop_handle_race.js", import.meta.url).href,
  { type: "module" },
);
