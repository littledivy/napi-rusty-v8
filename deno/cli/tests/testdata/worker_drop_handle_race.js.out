error: Uncaught (in worker "") Error
  throw new Error();
        ^
    at [WILDCARD]/workers/drop_handle_race.js:2:9
    at Object.action (deno:ext/timers/[WILDCARD])
    at handleTimerMacrotask (deno:ext/timers/[WILDCARD])
error: Uncaught (in promise) Error: Unhandled error event in child worker.
    at Worker.#pollControl (deno:runtime/js/11_workers.js:[WILDCARD])
